### PR TITLE
[DRAFT][Don't Review] test-case: add new test case for soc power measurement

### PR DESCRIPTION
**Pull request by fredoh9**
Intel SoCWatch is proprietary tool to measure with energy analsis for Intel platofrm. With socwatch, plan to check package C-State and average package power number etc.

Intel SoCWatch reference link
https://www.intel.com/content/www/us/en/docs/socwatch/get-started-guide/2020/overview.html

### CODE REVIEW

File: test-case/check-soc-power-status.sh [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+OPT_HAS_ARG['w']=1         OPT_VAL['w']=5
+
+# TODO: socwatch test after unloading audio module might be useful
+OPT_NAME['u']='unload-audio'  OPT_DESC['u']='unload audio modules for the test'
+OPT_HAS_ARG['u']=0            OPT_VAL['u']=0
+
+# TODO: every time should install socwatch? yes, must deploy everytime with built socwatch modules
+# FIXME: change to right path, probably ~/socwatch
+#SOCWATCH_PATH="$HOME/Downloads/socwatch_chrome_linux_INTERNAL_v2023.3.0_x86_64"
+SOCWATCH_PATH="$HOME/socwatch"
```
Comment (marc-hb) [inline test-case/check-soc-power-status.sh:43]:
```suggestion
# Set default value if none yet
: ${SOCWATCH_PATH:="$HOME"/socwatch}
```

File: test-case/check-soc-power-status.sh [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+# reference cmd: sudo ./socwatch -t 20 -s 5 -f cpu-cstate -f pkg-pwr -o fredtest5
+#SOCWATCH_CMD="./socwatch"
+SOCWATCH_FEATURE_PARAMS="-f cpu-cstate -f pkg-pwr"
```
Comment (marc-hb) [inline test-case/check-soc-power-status.sh:47]: Use a bash array for whitespace and special character safety
```suggestion
SOCWATCH_FEATURE_PARAMS=( -f cpu-cstate -f pkg-pwr )
```

Then expand with "${SOCWATCH_FEATURE_PARAMS[@]}"

File: test-case/check-soc-power-status.sh [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    
+    # set up checkpoint for each iteration
+    setup_kernel_check_point
+
+    # load socwatch module
+    sudo $SOCWATCH_PATH/insmod-socwatch
+    check_socwatch_module_loaded || die "socwatch module not loaded"
+
+    dlogc "Run the command: socwatch -t $duration -s $wait_time $SOCWATCH_FEATURE_PARAMS -o $SOCWATCH_PATH/sofsocwatch-$i"
+    sudo "$SOCWATCH_PATH"/socwatch -t "$duration" -s "$wait_time" $SOCWATCH_FEATURE_PARAMS -o "$SOCWATCH_PATH/sofsocwatch-$i" || die "socwatch returned $?"
```
Comment (marc-hb) [inline test-case/check-soc-power-status.sh:75]: See bash array above to fix shellcheck warning.

(and of course use shellcheck)

File: test-case/check-soc-power-status.sh [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+func_opt_parse_option "$@"
+func_lib_check_sudo
+duration=${OPT_VAL['d']}
+wait_time=${OPT_VAL['w']}
+loop_count=${OPT_VAL['l']}
+
+check_socwatch_module_loaded()
+{
+    sudo lsmod | grep -q socwatch || {
```
Comment (marc-hb) [inline test-case/check-soc-power-status.sh:57]:
```suggestion
    lsmod | grep -q socwatch ||
```
Don't need sudo.

Don't need `{` for a single statement (you can keep the newline)

File: test-case/check-soc-power-status.sh [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    dlogi "===== Loop($i/$loop_count) ====="
+    
+    # set up checkpoint for each iteration
+    setup_kernel_check_point
+
+    # load socwatch module
+    sudo $SOCWATCH_PATH/insmod-socwatch
+    check_socwatch_module_loaded || die "socwatch module not loaded"
+
+    dlogc "Run the command: socwatch -t $duration -s $wait_time $SOCWATCH_FEATURE_PARAMS -o $SOCWATCH_PATH/sofsocwatch-$i"
```
Comment (marc-hb) [inline test-case/check-soc-power-status.sh:74]: With a local `set -x` you don't need the copy/paste/out-of-sync.

```suggestion
    ( set -x
      sudo "$SOCWATCH_PATH"/socwatch -t "$duration" -s "$wait_time" $SOCWATCH_FEATURE_PARAMS -o "$SOCWATCH_PATH/sofsocwatch-$i"  ) ||
         die "socwatch returned $?"
```

**Review by marc-hb — COMMENTED** [review]
"check-soc-power-status.sh" sounds a bit like testing power usage is OK, which I don't think this test does.

I think a test which only makes sure SoCWatch is installed and works should have SoCwatch in the name. Maybe:
- test-case/run-socwatch.sh
- test-case/check-socwatch-works.sh

File: test-case/check-soc-power-status.sh [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+socwatch_test_once()
+{
+    local i="$1"
+    dlogi "===== Loop($i/$loop_count) ====="
+    
+    # set up checkpoint for each iteration
+    setup_kernel_check_point
+
+    # load socwatch module
+    sudo $SOCWATCH_PATH/insmod-socwatch
```
Comment (greg-intel) [inline test-case/check-soc-power-status.sh:71]: Shellcheck recommends quoting $SOCWATCH_PATH

**Comment by fredoh9** [comment]
Need to push more test commits, doesn't want to trigger device test every change. close this for now

**Comment by fredoh9** [comment]
Re-open, will ask for review

**Comment by fredoh9** [comment]
Re-open, will ask for review

**Comment by fredoh9** [comment]
it was  closed to test more internally. tried to reopen but github didn't allow to re-open. I didn't know the restriction. Will create new.

ImgPic_20230823_151801

**Pull request closed without merging**